### ISSUE TITLE
Add support for binlog version 9 file format

### ISSUE THREAD

**Issue by AdmiringWorm** [cake-contrib/Cake.Issues.MsBuild#178]
It would be great to have support for the latest binlog version 9 that gets outputted when using `/binarylogger` together with dotnet build (Assuming the latest version of MSBuild as well).

I am guessing this support would be included with the update with the PR #158.

**Comment by pascalberger** [comment]
I assume so, but Version 2.1 drops support for .NET 4.6 and MsBuild 14 and bumps requirements to .NET 4.7.2 and MsBuild to 15.1.0.

I personally can live with this restriction, even if it means that requirements for the addin are higher than what Cake itself requires and removes support for older Visual Studio versions. 

Opinions?

**Comment by AdmiringWorm** [comment]
I got nothing against it, but if it is of concern a fallback could be used.
By using the framework dependency restrictions.

For instance,
when restoring for .NET Core and .NET 4.7.2 use the 2.1 version, and for .NET 4.6 use the 2.0 version.

**Comment by AdmiringWorm** [comment]
This would of course require multi-targeting the csproj file though

**Comment by pascalberger** [comment]
Binlog version 9 support was added to MsBuildStructuredLog with https://github.com/KirillOsenkov/MSBuildStructuredLog/commit/11e6d4b45ea21d9d488e4f0cd1dfb58de5809662 in version 2.1.43

**Comment by AdmiringWorm** [comment]
@pascalberger do you know if it is possible to tell the binary logger on .NET Core to use a certain file format version?

I haven't found any information about this, but then again I barely found information about the binary logger at all.

**Comment by pascalberger** [comment]
I don't think this is possible. The version is hardcoded. See e.g. [here](https://github.com/Microsoft/msbuild/blob/master/documentation/wiki/Binary-Log.md)

**Comment by AdmiringWorm** [comment]
I was afraid of that.
I was just reading that documentation, and I think I found a different solution (that hopefully works).

**Comment by pascalberger** [comment]
Reopening since it doesn't seem to be possible to have an addin target .NET Framework 4.7.2 when using Cake runner for .NET Framework.

**Comment by GeertvanHorrik** [comment]
Hitting this one as well. Somehow the xml logger does not pick up my errors (only warnings), so was trying the binary log file but that uses version 9.

**Comment by pascalberger** [comment]
@GeertvanHorrik Parsing errors will be available in 0.9.1 (for both log formats. See https://github.com/cake-contrib/Cake.Issues.MsBuild/issues/201). I can push a release later today.

I don't see a possibility to support binlog version 9 without targeting a newer .NET version. I would like to avoid this in the Cake 1.x release cycle. As soon as Cake will bump its requirements (which will be a discussion for Cake 2.x) we can bump it here too. 

A workaround for using binlog support in Cake.Issues.MsBuild with never MsBuild versions is to explicitely set the logger, so that the binlog is written in a version which is compatible with Cake.Issues.MsBuild:

```
var msBuildSettings =
    new DotNetCoreMSBuildSettings()
        .WithLogger(
            "BinaryLogger," + Context.Tools.Resolve("Cake.Issues.MsBuild*/**/StructuredLogger.dll"),
            "",
            @"c:\mylog.bin");

DotNetCoreBuild(
    @"c:\mysolution.sln",
    new DotNetCoreBuildSettings
    {
        MSBuildSettings = msBuildSettings,
    });
```
